dialogue-core/src/main/java/com/palantir/dialogue/core/NeverThrowChannel.java

```diff
@@ -52,4 +61,27 @@
     public String toString() {
         return "NeverThrowChannel{" + delegate + '}';
     }
+
+    private static final class NeverThrowEndpointChannel implements EndpointChannel {
```

Comment:
I'm not a huge fan of all of this duplication :(

Reply:
Yeah this is because this class is used in both the per-uri 'channel' style (below the NSS) and also in the 'precompiled endpoint' style above the NSS.

I _think_ it might actually make sense to move the NeverThrow logic into the runtime, because actually if users provide a non-sensical Channel implementation it would actually be nice to still catch mistakes and turn them into failedFutures...